apollo-router/tests/integration/typename.rs

```diff
@@ -215,6 +209,9 @@ async fn two_named_operations() {
 async fn make_request(request: Request) -> apollo_router::graphql::Response {
     apollo_router::TestHarness::builder()
         .configuration_json(json!({
+            "supergraph": {
+                "introspection": true,
+            },
```

Comment:
To make sure. 
These tests don't work without this flag, right?
If the query only has `__typename`, is it treated as an introspection query?

Reply:
Yes without this these tests return `INTROSPECTION_DISABLED`

… but I didn’t realize opening this PR: that’s also a bug! Even if we internally use the introspection execution engine to handle `__typename` in fragment, it probably shouldn’t be rejected when (schema-)introspection is disabled

I’ll tweak the rejection logic in this PR